samples/database.sample.yml

```diff
@@ -0,0 +1,670 @@
+    - category:   VTA-2
+      definition: s/he puts the lights on for s.o.
+    - category:   VTA-2
+      definition: s/he lights (it/him) for s.o.
+  specificWordClass: VTA
+
+- fstStem:          mâcisw
+  generalWordClass: V
+  key:              mâciswêw
+  lemma:
+    sro: mâciswêw
+    cw:  mâciswêw
+  senses:
+    - category:   VTA-3
+      definition: s/he cuts s.o. open
+  specificWordClass: VTA
+
+- fstStem:          kispêwât
+  generalWordClass: V
+  key:              kispêwâtêw
+  lemma:
+    sro: kispêwâtêw
+    cw:  kispêwâtêw
+  senses:
+    - category:   VTA-4
+      definition: s/he defends s.o., s/he is overprotective of s.o.
+  specificWordClass: VTA
+
+- fstStem:          ay-it
+  generalWordClass: V
+  key:              ay-itêw
+  lemma:
+    sro: ay-itêw
+    cw:  ay-itêw
+  senses:
+    - category:   VTA-5
+      definition: s/he says so to s.o, s/he says so of s.o.
+  specificWordClass: VTA
+
+- fstStem:          kâhtin
+  generalWordClass: V
+  key:              kâhtinam
+  lemma:
+    sro: kâhtinam
+    cw:  kâhtinam
+  senses:
+    - category:   VTI-1
+      definition: s/he obtains s.t., s/he attains s.t., s/he procures s.t.
+    - category:   VTI-1
+      definition: s/he pushes s.t.
+  specificWordClass: VTI
+
+- fstStem:          nahêkâcihtâ
+  generalWordClass: V
+  key:              nahêkâcihtâw
+  lemma:
+    sro: nahêkâcihtâw
+    cw:  nahêkâcihtâw
+  senses:
+    - category:   VTI-2
+      definition: s/he is sparing of s.t.
+  specificWordClass: VTI
+
+- fstStem:          oscoscocasi
+  generalWordClass: V
+  key:              oscoscocasiw
+  lemma:
+    sro: oscoscocasiw
+    cw:  oscoscocasiw
+  senses:
+    - category:   VTI-3
+      definition: s/he coughs a little
+    - category:   VTI-3
+      definition: s/he coughs up a little of something
+  specificWordClass: VTI
+
+- fstStem:          atimw
+  generalWordClass: N
+  key:              atim
+  lemma:
+    sro: atim
+    cw:  atim
+  senses:
+    - category:   NA-3
+      definition: dog
+    - category:   NA-3
+      definition: horse
+    - category:   NA-3
+      definition: beast of burden
+  forms:
+    - transcription:
+        sro: têm
+        cw:  têm
+    - transcription:
+        sro: atimw
+        cw:  atimw
+  specificWordClass: NA
+
+- fstStem:          atâhkw
+  generalWordClass: N
+  key:              atâhk
+  lemma:
+    sro: atâhk
+    cw:  atâhk
+  senses:
+    - category:   NA-3
+      definition: star
+  specificWordClass: NA
+
+- fstStem:          acâhkos
+  generalWordClass: N
+  key:              acâhkos
+  lemma:
+    sro: acâhkos
+    cw:  acâhkos
+  senses:
+    - category:   NA-1
+      definition: star, little star
+  specificWordClass: NA
+
+- fstStem:          nîmi
+  generalWordClass: V
+  key:              nîmiw
+  lemma:
+    sro: nîmiw
+    cw:  nîmiw
+  senses:
+    - category:   VAI-1
+      definition: s/he dances
+  specificWordClass: VAI
+
+- generalWordClass: N
+  key:              nîminâniwan
+  lemma:
```

Comment:
Based on our discussions, where we've wanted to avoid ambiguity, my understanding has been that the term (FST)-lemma would throughout be the representative word-form for the lexeme, whether in our  external dictionary database or within iDict, or for FST analysis/generation. Here, that would be `nîmiw,` based on the FST analysis: `nîmiw+V+AI+Ind+X`. This way, this entry, while having an analysis and static definition, would point to the FST-lemma `nîmiw`, for which hopefully there is an entry which would also be presented, and which would be the entry point for accessing the paradigm.

A question of its own is whether such an FST-lemma is provided beforehand in the dictionary database that is imported, or whether that is assigned dynamically during the importation. I'd be inclined to prefer the latter. Or perhaps this might be a two-step process where we use iDict-internally an FST-lemma, if such is provided for import, but if not, we try FST-analysis to dynamically establish the FST-lemma, and if the FST-analysis is unsuccessful, then the FST-lemma will be set to NULL.

Reply:
After reading this and discussing with @andrewdotn this morning, I'll be explicitly indicating the FST lemma now (pending the update to this PR I'm working on).

Reply:
The reason I raised the possibility of assigning the (FST-)lemma dynamically rather than requiring that in whatever formatted dictionary source we request from a collaborating linguist is that that task could be taken automatically, when having an FST and perhaps specifying some clear rules how the (FST-)lemma is assigned (or left unassigned as something like NULL), and relieving the linguist from having to do that.

In all simplicity the FST lemma could be assigned to an entry, if the FST-analysis of the entry head is successful, from which analysis we can extricate the lemma. If the FST-analysis is not successful, then both the analysis and lemma fields could be empty/NULL.

Of course, the challenge to this is that there will be a number of ambiguous cases where it may not be trivial to resolve language-independently which analysis, and which lemma, applies to the entry head in question. This likely could be achieved by language specific rules, but this should perhaps not be the developer's primary problem - but rather the linguist's headache.

Of course, the assignment of FST lemma could be kept as a separate functionality preceding the importation, requiring two actions (1. assign FST lemmas to imported dictionary source, if desired -> 2. import dictionary source) - and we could make available a generic solution. To the linguist this would necessarily not make a difference.